### PR TITLE
Adding secrets support to `firebase deploy --only extensions`

### PR DESCRIPTION
**Pull request by joehan**
### Description
Adding secrets support to `firebase deploy --only extensions`. 
Unit tests for the new methods in secrets.ts to follow - I wanted to get some eyes on this before it got too late in the day/

The logic here can be a little hard to follow - to help, I made a flow chart: go/firex-deploy-secrets-diagram

### Scenarios Tested
I tested each of the cases in go/firex-deploy-secrets-diagram using a bunch of instances of pavelj/secrets-test@0.0.11 & 0.0.10
Screen Shot 2021-10-21 at 10 55 09 AM

### CODE REVIEW

File: src/deploy/extensions/secrets.ts [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+  secretVersion?: secretManager.SecretVersion;
+  labels: Record<string, string>;
+}> {
+  const secretInfo: any = {
+    labels: {},
+  };
+  try {
+    secretInfo.secret = await secretManager.getSecret(projectId, secretName);
+    secretInfo.secretVersion = await secretManager.getSecretVersion(projectId, secretName, version);
+    secretInfo.labels = await secretManager.getSecretLabels(projectId, secretName);
```
Comment (pavelgj) [inline src/deploy/extensions/secrets.ts:204]: optional: getSecretLabel is identical to getSecret two lines above, it just only returns labels and the getSecret doesn't include labels... we should really combine those two....

Comment (pavelgj) [reply src/deploy/extensions/secrets.ts:204]: I mean, it's my own fault... for implementing it this way, I just didn't need those two in the same place.... :)

File: src/deploy/extensions/secrets.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    throw new FirebaseError(
+      `${clc.bold(i.instanceId)}: Found '${providedValue}' for secret param ${
+        secretParam.param
+      }. ` +
+        `projects/${projectId}/secrets/${secretName} is managed by a different extension instance (${
+          secretInfo.labels[secretUtils.SECRET_LABEL]
+        }), so reusing it here can lead to unexpected behavior. ` +
+        "Please choose a different name for this secret, and rerun this command."
+    );
+  }
```
Comment (pavelgj) [inline src/deploy/extensions/secrets.ts:133]: If secret/version exist and don't have the label we just use it, right? I think you still need to call `secretsUtils.grantFirexServiceAgentSecretAdminRole(secretInfo.secret)` on it so that we get all the necessary P4SA permissions. Similar in the update case.

**Review by pavelgj — CHANGES_REQUESTED** [review]

File: src/deploy/extensions/secrets.ts (hunk 1 above)
Comment (joehan) [reply src/deploy/extensions/secrets.ts:204]: Haha, I noticed this and made a mental note to fix it later, then forgot to fix it later. I'll dedupe this logic
